sample-code/vc-using-framestream/vc-using-framestream/ArisRecording.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "Frame.h"
+#include <memory>
+#include <fstream>
+
+class WriteableArisRecording
+{
+public:
+  // Creates a writable recording file. Returns an unassociated unique_ptr on
+  // failure. If no frames are written the file is deleted in the destructor.
+  // If a file already exists it is overwritten.
+  static std::unique_ptr<WriteableArisRecording> Create(const char * path);
+
+  // Use Create() rather than the constructor.
+  WriteableArisRecording(const char * path, std::ofstream && initializedOutfile);
```

Comment:
Consider making the constructor private so consumers can't get themselves into trouble.